opencog/atoms/core/ScopeLink.cc

```diff
@@ -313,29 +358,23 @@ ContentHash ScopeLink::term_hash(const Handle& h,
 		const Variables& vees = sco->get_variables();
 		for (const Handle& v : vees.varseq) bound_vars.insert(v);
 	}
-
-	// Prevent mixing for UnorderedLinks. The `mixer` var will be zero
-	// for UnorderedLinks. The problem is that two UnorderdLinks might
-	// be alpha-equivalent, but have their atoms presented in a
-	// different order. Thus, the hash must be computed in a purely
-	// commutative fashion: using only addition, so as to never create
-	// any entropy, until the end.
-	//
-	// XXX As discussed in issue #1176, a better fix would be to
-	// compute the individual term_hashes first, then sort them,
-	// and then mix them!  This provides the desired qualities:
-	// different unordered links can be directly compared, and also
-	// have good mixing/avalanching properties. The code below
-	// only allows for compare; it fails to mix.
-	//
-	bool is_ordered = not classserver().isA(t, UNORDERED_LINK);
-	ContentHash mixer = (ContentHash) is_ordered;
+	// As discussed in issue #1176, compute the individual term_hashes
+	// first, then sort them, and then mix them!  This provides the
+	// desired qualities: different unordered links can be directly
+	// compared, and also have good mixing/avalanching properties.
 	ContentHash hsh = ((1UL<<8) - 59) * t;
+	std::vector<ContentHash> hash_vec;
 	for (const Handle& ho: h->getOutgoingSet())
 	{
-		hsh += mixer * (hsh<<5) + term_hash(ho, bound_vars, quotation);
+		hash_vec.push_back(term_hash(ho, bound_vars, quotation));
+	}
+	// hash_vec should be sorted only for unordered links
+	if (classserver().isA(t, UNORDERED_LINK)) {
+		std::sort(hash_vec.begin(), hash_vec.end());
```

Comment:
@linas said
> Specifically, the actual sort order used should be identical to that in the constructors in `atoms/core/UnorderedLink.cc`

I don't see why it would need to be identical, at the very least it only needs to be compatible, when the hashes are different, and obviously it is, since `Link::operator<` and `Node::operator<` rely on the hashes, when they are different. But again, even that doesn't seem necessary, it's just to compute the hash of the scope, nothing else.